site/components/pages/Pledge/PledgeDashboard/index.tsx

```diff
@@ -42,6 +42,18 @@ export const PledgeDashboard: NextPage<Props> = (props) => {
     setShowModal(false);
   };
 
+  /**
+   * Close the modal on escape press
+   * @todo extract as generic modal behaviour
+   */
+  useEffect(() => {
+    const escListener = (e: KeyboardEvent) =>
+      e.key === "Escape" && setShowModal(false);
```

Comment:
I would check first if the modal is open 
`showModal && e.key === "Escape" && setShowModal(false);`

Reply:
Sure thing

Reply:
Sure thing